### PR TITLE
feat: refactor four attention converters with TRT `add_attention_v2()`

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

This PR refactors four attention converters with TRT `add_attention_v2()`. Additionally, since newer torch rejects passing both `is_causal=True` and explicit `attn_mask` to SDPA, which aligns with TRT, some scenarios are gated in validators and the unreachable branches are pruned.

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/models/test_hf_gqa_model.py [hunk 1]
```diff
@@ -82,4 +82,4 @@ def test_dynamic_head_dim_with_hf_model(dtype, decompose_attention):
         elif isinstance(out, (tuple, list)):
             out = out[0]
 
-    torch.testing.assert_close(ref, out, rtol=1e-1, atol=2e-1)
+    torch.testing.assert_close(ref, out, rtol=0.3, atol=0.3)
```
Comment (narendasan) [inline tests/py/dynamo/models/test_hf_gqa_model.py]: This is quite high? Will it be stable in practice?

Comment (zewenli98) [reply tests/py/dynamo/models/test_hf_gqa_model.py]: The relaxation is mainly for TRT-RTX. As I talked with @SandSnip3r, TRT-RTX 1.6 is going to fix the accuracy issue but it's not released yet. So I skipped these tests for now and changed to `rtol=1e-1, atol=3e-1` for the single slightly mismatched element on standard TRT:
```
Mismatched elements: 1 / 9723904 (0.0%)
Greatest absolute difference: 0.2071533203125 at index (0, 0, 80751) (up to 0.2 allowed)
Greatest relative difference: 2.931640625 at index (0, 0, 80751) (up to 0.1 allowed)
```

File: py/torch_tensorrt/dynamo/conversion/impl/attention.py [hunk 2]
```diff
@@ -69,6 +68,60 @@ def _normalize_attention_mask_rank(
     return mask
 
 
+def _restore_static_query_heads(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/impl/attention.py:71]: Should this be a lowering pass?

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/attention.py:71]: I'm not sure if this is general enough to make it a lowering pass now because I only observed in the `query` of attention in some model implementations like [HF Qwen](https://github.com/huggingface/transformers/blob/bb3ffb9703e3acb84f06db1d3799756e977662c2/src/transformers/models/qwen3/modular_qwen3.py#L79). Maybe we can keep it in the attention converters now. If we observe more cases later, we can make it a lowering pass then.

**Review by narendasan — APPROVED** [review]
LGTM
